src/services/render/index.tsx

```diff
@@ -80,24 +75,26 @@ export const renderElement = (
     );
 
     if (Component) {
-      /* eslint-disable max-len */
-      // Use object spreading instead of explicitly setting the key (to potentially undefined values)
-      // Explicitly setting the key causes collision when several components render with `undefined` value for `key`
-      // Object spreading will define the prop only when its value is truthy
-      /* eslint-enable max-len */
-      const extraProps: Partial<{ [key: string]: string | null }> = {};
-      if (element.getAttribute('key')) {
-        extraProps.key = element.getAttribute('key');
+      // Prepare props for the component
+      const props = {
+        element,
+        onUpdate,
+        options: {
+          ...options,
+          inlineFormattingContext,
+        },
+        stylesheets,
+      };
+
+      // Conditionally render the component with a key if it exists, to avoid
+      // warnings with current React versions, when the key attribute is set
+      // using the spread operator.
+      const key = element.getAttribute('key');
+
+      if (key) {
+        <Component {...props} key={key} />; // eslint-disable-line react/jsx-props-no-spreading
```

Comment:
@fbinz I think you've accidentally removed the return statement here

Reply:
Oh, damn! Glad you caught that. Sorry for the troubles.

Reply:
thanks for the fix!